pkg/codegen/dotnet/gen_program_expressions.go

```diff
@@ -832,7 +849,11 @@ func (g *generator) genObjectConsExpressionWithTypeName(
 				lit := item.Key.(*model.LiteralValueExpression)
 				propertyKey := lit.Value.AsString()
 				g.Fprint(w, resolvePropertyName(propertyKey, propertyNames))
+				if g.usingDefaultListInitializer() && isEmptyList(item.Value) {
+					g.listInitializer = "new()"
+				}
 				g.Fgenf(w, " = %.v,\n", item.Value)
+				g.resetListInitializer()
```

Comment:
Could this causes issues with lists in lists?

Reply:
I don't think so but I am not 100% sure either so I changed it such that it doesn't modify the state
```go
if g.usingDefaultListInitializer() && isEmptyList(item.Value) {
	g.Fgen(w, " = new() { },\n")
} else {
	g.Fgenf(w, " = %.v,\n", item.Value)
}
```
To be honest, this is far from ideal. All because we don't have context information about the full type of the array at `GenTupleConsExpression` 😤